src/progly/run-papertests.cc

```diff
@@ -0,0 +1,312 @@
+  res_file << res << "\n" ;
+  res_file.close() ;
+
+  return 0 ;
+}
+
+void raw_whole_write( std::string oid, std::string filename, std::string poolname, bool debug ) {
+
+  std::ifstream infile ;
+  infile.open( filename ) ;
+
+  std::string line ;
+  librados::bufferlist bl ;
+  while( std::getline( infile, line ) ) {
+    ::encode( line, bl ) ;
+  }
+
+  // save to ceph object
+  // connect to rados
+  librados::Rados cluster;
+  cluster.init(NULL);
+  cluster.conf_read_file(NULL);
+  int ret = cluster.connect();
+
+  // open pool
+  librados::IoCtx ioctx ;
+  ret = cluster.ioctx_create( poolname.c_str(), ioctx ) ;
+
+  // write bl_seq to ceph object
+  const char *obj_name = oid.c_str() ;
+  bufferlist::iterator p = bl.begin();
+  size_t num_bytes_written = p.get_remaining() ;
+  if( debug )
+    std::cout << "num_bytes_written : " << num_bytes_written << std::endl ;
+  ret = ioctx.write( obj_name, bl, num_bytes_written, 0 ) ;
+
+  ioctx.close() ;
+}
+
+void raw_iter_write( std::string oid, std::string filename, std::string poolname, bool debug ) {
+
+  std::ifstream infile ;
+  infile.open( filename ) ;
+
+  std::string line ;
+  librados::bufferlist bl ;
+  while( std::getline( infile, line ) ) {
+    //line.erase( std::remove( line.begin(), line.end(), '\n' ), line.end() ) ;
+    ::encode( line, bl ) ;
+  }
+
+  // save to ceph object
+  // connect to rados
+  librados::Rados cluster;
+  cluster.init(NULL);
+  cluster.conf_read_file(NULL);
+  int ret = cluster.connect();
+
+  // open pool
+  librados::IoCtx ioctx ;
+  ret = cluster.ioctx_create( poolname.c_str(), ioctx ) ;
+
+  // write bl_seq to ceph object
+  const char *obj_name = oid.c_str() ;
+  bufferlist::iterator p = bl.begin();
+  size_t num_bytes_written = p.get_remaining() ;
+  if( debug )
+    std::cout << "num_bytes_written : " << num_bytes_written << std::endl ;
+  ret = ioctx.write( obj_name, bl, num_bytes_written, 0 ) ;
+
+  ioctx.close() ;
+}
+
+void do_sum( librados::bufferlist bl_seq, int write_type, bool debug ) {
+
+  std::vector< uint64_t > these_ints ;
+
+  // raw
+  if( write_type == 0 ) {
+    ceph::bufferlist::iterator it_bl_seq = bl_seq.begin() ;
+    ceph::bufferlist bl ;
+    ::decode( bl, it_bl_seq ) ;
+    const char* raw = bl.c_str() ;
+    int count = 0 ;
+    std::string tmp = "" ;
+
+    for( int i = 0; i < std::strlen( raw ); i++ ) {
+      tmp = tmp + raw[i] ;
+      count++ ;
+      if( count == 0 ) {
+        continue ;
+      }
+      else if( count % 6 == 0 ) {
+        if( debug )
+          std::cout << "appending tmp = " << tmp << std::endl ;
+        these_ints.push_back( std::stoi( tmp ) ) ;
+        count = 0 ;
+        tmp = "" ;
+      }
+    }
+  }
+  // raw indiv
+  else if( write_type == 1 ) {
+    ceph::bufferlist::iterator it_bl_seq = bl_seq.begin() ;
+    while( it_bl_seq.get_remaining() > 0 ) {
+      ceph::bufferlist bl ;
+      ::decode( bl, it_bl_seq ) ;
+      const char* raw = bl.c_str() ;
```

Comment:
if these ints were orginally encoded directly into bl, you can directly decode into a int here, to avoid decoding into a bl then using bl.c_str*() and then stoi().

Reply:
done. also using your recommendation of pop_back to strip the newline.